app/javascript/controllers/select_gif_controller.js

```diff
@@ -10,8 +10,16 @@ export default class extends Controller {
     if (inputElement) {
       inputElement.value = this.urlValue;
 
+
       const previousGifElement = document.querySelector(".individual-gif.border-4.border-blue-500");
 
+      // De-select if selecting the same gif
+      if (previousGifElement && previousGifElement === this.gifTarget) {
+        previousGifElement.classList.remove("border-4", "border-blue-500");
+        inputElement.value = "";
```

Comment:
This is a good idea to deselect as well so that search can be made again if needed 👌